src/model/interactions.rs

```diff
@@ -0,0 +1,108 @@
+//! Interactions information-related models
+
+use super::prelude::*;
+use crate::internal::prelude::*;
+
+/// Information about an Interaction
+///
+/// An interaction is the base "thing" that is sent when a user invokes a command, and is the same
+/// for Slash Commands and other future interaction types.
+#[derive(Clone, Debug, Deserialize, Serialize)]
+#[non_exhaustive]
+pub struct Interaction {
+    pub id: InteractionId,
+    #[serde(rename = "type")]
+    pub kind: InteractionType,
+    pub data: Option<ApplicationCommandInteractionData>,
+    pub guild_id: GuildId,
+    pub channel_id: ChannelId,
+    pub member: Member,
+    pub token: String,
+    pub version: i32
+}
+
+/// The type of an Interaction
+#[derive(Copy, Clone, Debug, Deserialize, Hash, Eq, PartialEq, PartialOrd, Ord, Serialize)]
+#[non_exhaustive]
+#[repr(u8)]
+pub enum InteractionType {
+    Ping = 1,
+    ApplicationCommand = 2
+}
+
+/// The command data payload
+#[derive(Clone, Debug, Deserialize, Serialize)]
+#[non_exhaustive]
+pub struct ApplicationCommandInteractionData {
+    pub id: CommandId,
+    pub name: String,
+    #[serde(default)]
+    pub options: Vec<ApplicationCommandInteractionDataOption>
+}
+
+/// A set of a param and value from the user.
+///
+/// All options have names, and an option can either be a parameter and input value--in which case
+/// `value` will be set--or it can denote a subcommand or group--in which case it will contain a
+/// top-level key and another array of `options`.
+#[derive(Clone, Debug, Deserialize, Serialize)]
+#[non_exhaustive]
+pub struct ApplicationCommandInteractionDataOption {
+    pub name: String,
+    pub value: Option<ApplicationCommandOptionType>,
+    #[serde(default)]
+    pub options: Vec<ApplicationCommandInteractionDataOption>
+}
+
+/// The base "command" model that belongs to an application.
```

Comment:
```suggestion
/// The base command model that belongs to an application.
```
Do you mean *slash command* or why did you put *"command"*?